### PR TITLE
Added folder structure, Also added some files

### PR DESCRIPTION
**Pull request by Samvdg**
# Athena Back-end Setup

## Changes

- Added for back-end folders.

- Added MongoDB

- Added Express

### How to test:

- Pull from ```#9-backend-folder-structure``` check for linting errors & for clean code.

#### Note
```(node:5532) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.```

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const User = require ('../models/user.model');
+
+// simple version, no filtering
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// this is an example function, the exports.<function name here> is basicly the function itself
+
+// users/test
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+exports.test = function (req, res) {
+    res.send('Greetings from the Test controller!');
+};
+
+// users/create
+exports.users_create = (req, res) => {
+
+    // creating the user to send to the model
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+// simple version, no filtering
+// this is an example function, the exports.<function name here> is basicly the function itself
+
+// users/test
+exports.test = function (req, res) {
+    res.send('Greetings from the Test controller!');
+};
+
+// users/create
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    // creating the user to send to the model
+    let user = new User(
+        {
+            UserName: req.body.name,
+            UserPass: req.body.password,
+            UserEmail: req.body.email
+        }
+    );
+    // sending (and catching the error of sending) the created user to the model for insertion
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    user.save( (err) => {
+        if (err) {
+            return next(err);
+        }
+        // what it returns :D
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        // note to self: figure out how to turn this to a nice html acceptable form of the return value of the created user
+        res.send('User Created successfully')
+    })
+};
+
+// read functionality
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+exports.users_details = (req, res) => {
+    User.findById(req.params.id, (err, user) => {
+        if (err) return next(err);
+        res.send(user);
+    })
+}
+
+// update functionality
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: Capital letter at  beginning of sentence

File: src/controllers/userController.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    });
+};
+
+// delete functionality
+exports.users_delete = (req, res) => {
+    User.findByIdAndRemove(req.params.id, (err) => {
+        if (err) return next(err);
+        res.send('Deleted successfully!');
+    })
+};
```
Comment (MiguelFieira) [inline src/controllers/userController.js]: endline

File: src/models/user.model.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+// creating a database table
```
Comment (MiguelFieira) [inline src/models/user.model.js]: Capital letter at  beginning of sentence

File: src/models/user.model.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    UserName: {type: String, required: true, max: 25},
+    UserPass: {type: String, required: true, max: 100},
+    UserEmail: {type: String, required: true, max: 30},
+});
+
+// export
+module.exports = mongoose.model('user', UserSchema);
+
+// MVC is eigenlijk gewoon 1 groot bestand (app.js) uit elkaar trekken en het MVC gedeelte is eigenlijk gewoon de structuur van 
+// 1 groote functionaliteit gestructureerd verdelen over meerdere bestanden, dus MVC is de manier om deze te ordenen en bij te houden
```
Comment (MiguelFieira) [inline src/models/user.model.js]: endline

File: src/models/user.model.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+const Schema = mongoose.Schema;
+
+// creating a database table
+let UserSchema = new Schema({
+    UserName: {type: String, required: true, max: 25},
+    UserPass: {type: String, required: true, max: 100},
+    UserEmail: {type: String, required: true, max: 30},
+});
+
+// export
```
Comment (MiguelFieira) [inline src/models/user.model.js]: Capital letter at  beginning of sentence

File: src/routes/user.route.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+const router = express.Router();
+
+// Require the controllers
+// this is the file which contains all your functions (the crud of the API) that connect to the model
+const UserController = require ('../controllers/userController');
+
+// a simple test url to check that all of our files are communicating correctly.
```
Comment (MiguelFieira) [inline src/routes/user.route.js]: Capital letter at  beginning of sentence

File: src/routes/user.route.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+// this is the second slash within the uri, it contains the name of the function (ex. `test`) and uses the corresponding function via the immported
+// controller.
+// the router.<request here> can be for ex. post or get
+router.get('/test', UserController.test);
+router.post('/create', UserController.users_create);
+
+// the `:` infront of 'id' makes sure anything can go there, the `:` make sure its a param
```
Comment (MiguelFieira) [inline src/routes/user.route.js]: Capital letter at  beginning of sentence

File: src/routes/user.route.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+// updating an existing record
+router.put('/:id/update', UserController.users_update);
+
+// deleting an existing record
+router.delete('/:id/delete', UserController.users_delete);
+
+
+// export
+module.exports = router;
```
Comment (MiguelFieira) [inline src/routes/user.route.js]: endline

File: src/routes/user.route.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+router.get('/test', UserController.test);
+router.post('/create', UserController.users_create);
+
+// the `:` infront of 'id' makes sure anything can go there, the `:` make sure its a param
+router.get('/:id', UserController.users_details);
+
+// updating an existing record
+router.put('/:id/update', UserController.users_update);
+
+// deleting an existing record
```
Comment (MiguelFieira) [inline src/routes/user.route.js]: Capital letter at beginning of sentence

File: src/routes/user.route.js [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+// this is the second slash within the uri, it contains the name of the function (ex. `test`) and uses the corresponding function via the immported
+// controller.
+// the router.<request here> can be for ex. post or get
+router.get('/test', UserController.test);
+router.post('/create', UserController.users_create);
+
+// the `:` infront of 'id' makes sure anything can go there, the `:` make sure its a param
+router.get('/:id', UserController.users_details);
+
+// updating an existing record
```
Comment (MiguelFieira) [inline src/routes/user.route.js]: Capital letter at  beginning of sentence

File: app.js [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+
+// import routes for users
+// place the functions (the route files) in here (the first slash, ex. `users`, `posts' etc. which acces the functions within the corresponding 
+// controller via the second slash containing the function name ex. `test`)
+
+app.use('/users', user);
+
+// just a testing localhost thingy
```
Comment (MiguelFieira) [inline app.js]: Capital letter at beginning of sentence

File: app.js [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+mongoose.Promise = global.Promise;
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+// parse request bodies
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+
+// import routes for users
+// place the functions (the route files) in here (the first slash, ex. `users`, `posts' etc. which acces the functions within the corresponding
```
Comment (MiguelFieira) [inline app.js]: Capital letter at beginning of sentence

File: app.js [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+// Set up mongoose connection
+const mongoose = require('mongoose');
+let dev_db_url = 'mongodb://Crivian:floortje2001@ds119024.mlab.com:19024/athena_alpha'; // test url with mongoDB and mLab
+let mongoDB = process.env.MONGODB_URI || dev_db_url; 
+mongoose.connect(mongoDB);
+mongoose.Promise = global.Promise;
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+// parse request bodies
```
Comment (MiguelFieira) [inline app.js]: Capital letter at beginning of sentence

File: app.js [hunk 21]
```diff
@@ -0,0 +1,33 @@
+// app.js
+
+const express = require('express');
+const bodyParser = require('body-parser');
+
+const user = require ('./src/routes/user.route');
+const app = express();
+
+// initialize our express app
```
Comment (MiguelFieira) [inline app.js]: Capital letter at beginning of sentence

File: app.js [hunk 22]
```diff
@@ -0,0 +1,33 @@
+// app.js
```
Comment (MiguelFieira) [inline app.js]: For what is this comment?

**Review by MiguelFieira — CHANGES_REQUESTED** [review]
See comments

File: .gitignore [hunk 23]
```diff
@@ -3,3 +3,6 @@
 
 # Distribution 
 /dist
+
+# PhpStorm
+.idea
```
Comment (MiguelFieira) [inline .gitignore]: endline

**Review by MiguelFieira — CHANGES_REQUESTED** [review]
See Comment

File: src/controllers/userController.js [hunk 24]
```diff
@@ -0,0 +1,55 @@
+const User = require ('../models/user.model');
+
+// Simple version, no filtering
+// This is an example function, the exports.<function name here> is basicly the function itself
+
+// Users/test
+exports.test = function (req, res) {
+    res.send('Greetings from the Test controller!');
```
Comment (AjobK) [inline src/controllers/userController.js:8]: Make this a little more clear. If you have 100 controllers with this test message, we won't know what's what.

File: src/models/user.model.js [hunk 25]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+// Creating a database table
+let UserSchema = new Schema({
+    UserName: {type: String, required: true, max: 25},
+    UserPass: {type: String, required: true, max: 100},
```
Comment (AjobK) [inline src/models/user.model.js:7]: UserUsername
UserPassword
UserMail
UserPhone
UserRole

File: src/controllers/userController.js [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+        }
+    );
+    // Sending (and catching the error of sending) the created user to the model for insertion
+    user.save( (err) => {
+        if (err) {
+            return next(err);
+        }
+        // What it returns :D
+        // Note to self: figure out how to turn this to a nice html acceptable form of the return value of the created user
+        res.send('User Created successfully')
```
Comment (AjobK) [inline src/controllers/userController.js:29]: Please for gods sake do not 'Do This Kind Of Lettering And Use An Uppercase Letter For Every Word'

File: src/models/user.model.js (hunk 25 above)
Comment (AjobK) [reply src/models/user.model.js:7]: Userphone is optional

**Review by AjobK — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**